src/main/java/com/xjeffrose/xio/http/Http1ProxyHandler.java

```diff
@@ -11,16 +11,20 @@
 @Slf4j
 public class Http1ProxyHandler extends SimpleChannelInboundHandler<HttpObject> {
 
-  private final UrlRouter router;
-  private RouteProvider route;
-  private RouteUpdateProvider updater;
+  private final Router router;
+  private RequestHandler route;
+  private RequestUpdateHandler updater;
 
-  public Http1ProxyHandler(UrlRouter router) {
+  public Http1ProxyHandler(Router router) {
     this.router = router;
   }
 
   @Override
   public final void channelRead0(final ChannelHandlerContext ctx, HttpObject msg) throws Exception {
+    if (router == null) {
+      // Router is null meaning it has not been set yet.
+      throw new IllegalStateException("No Router");
+    }
```

Comment:
Don't do this. ApplicationState.getRouter should be able to provide this handler a router at pipeline construction time. If you absolutely cannot get a router configuration, instantiate a NullRouter instance. I'm of the opinion that there should be a sane default router configuration that we can load at startup.

Reply:
i can totally create a default or null router for this.   it minimum it should at least log a warning so that the logs tell the tale.   as for sane default router config, it would be hard because all this comes from the registry and we can make no assumptions about what that config might be...   for now i'll create a default router and just log usage...